File: tools/src/main/java/com/jiangkang/tools/utils/AppUtils.java
package com.jiangkang.tools.utils;

import android.app.Activity;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.util.Log;

import com.jiangkang.tools.King;

import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

/**
 * Created by jiangkang on 2017/9/8.
 *
 * 与App相关的工具类
 */

public final class AppUtils {

    private static final String TAG = AppUtils.class.getSimpleName();

    public static String getAppVersionName() {
        try {
            PackageInfo packageInfo = getPackageInfo();
            return packageInfo.versionName;
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        return "unknown";
    }


    private static PackageInfo getPackageInfo() throws PackageManager.NameNotFoundException {
        PackageManager pm = King.getApplicationContext().getPackageManager();
        return pm.getPackageInfo(King.getApplicationContext().getPackageName(),PackageManager.GET_CONFIGURATIONS);
    }



    public static String getCurrentActivity(){
        return King.getTopActivityWeakRef().get().getComponentName().getClassName();
    }


    public static List<String> getActivityListInStack(){
        List<Activity> activities = King.getsActivityList();
        List<String> list = new ArrayList<>();
        Iterator iterator = activities.iterator();
        while (iterator.hasNext()){
            Activity activity = (Activity) iterator.next();
            list.add(activity.getComponentName().getClassName());
        }
        return list;
    }






}

File: tools/src/main/java/com/jiangkang/tools/utils/AppUtils.kt
package com.jiangkang.tools.utils

import android.app.Activity
import android.content.pm.ApplicationInfo
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import com.jiangkang.tools.King
import java.util.*

/**
 * Created by jiangkang on 2017/9/8.
 *
 * 与App相关的工具类
 */

object AppUtils {

    private val TAG = AppUtils::class.java.simpleName
    

    val appVersionName: String
        get() {
            try {
                val packageInfo = packageInfo
                return packageInfo.versionName
            } catch (e: PackageManager.NameNotFoundException) {
                e.printStackTrace()
            }
            return "unknown"
        }


    private val packageInfo: PackageInfo
        @Throws(PackageManager.NameNotFoundException::class)
        get() {
            val pm = King.getApplicationContext().packageManager
            return pm.getPackageInfo(King.getApplicationContext().packageName, PackageManager.GET_CONFIGURATIONS)
        }


    val currentActivity: String
        get() {
            return King.getTopActivityWeakRef().get()!!.componentName.className
        }


    val activityListInStack: List<String>
        get() {
            val activities = King.getsActivityList()
            val list = ArrayList<String>()
            val iterator = activities.iterator()
            while (iterator.hasNext()) {
                val activity = iterator.next() as Activity
                list.add(activity.componentName.className)
            }
            return list
        }


    val isDebug: Boolean
        get() {
            val appInfo: ApplicationInfo = King.getApplicationContext().applicationInfo
            if ((appInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE) != 0) {
                return true
            }
            return false
        }


}
